command/agent/srv.go

```diff
@@ -0,0 +1,114 @@
+package agent
+
+import (
+	"fmt"
+	"io"
+	"log"
+	"net"
+	"time"
+)
+
+const (
+	srvPollInterval = 60 * time.Second
+	srvFindTimeout  = srvPollInterval / 2
+)
+
+// AgentSRV periodically polls an SRV record
+// And attempts to join the hosts supplied
+type AgentSRV struct {
+	agent      *Agent
+	srvrecords []string
+	logger     *log.Logger
+	replay     bool
+}
+
+// NewAgentSRV is used to create a new AgentSRV
+func NewAgentSRV(agent *Agent, logOutput io.Writer, replay bool, srvrecords []string) (*AgentSRV, error) {
+
+	// Initialize the AgentSRV
+	m := &AgentSRV{
+		agent:      agent,
+		srvrecords: srvrecords,
+		logger:     log.New(logOutput, "", log.LstdFlags),
+		replay:     replay,
+	}
+
+	// Start the poller in the background
+	go m.run()
+	return m, nil
+}
+
+// run is a long running goroutine that scans for new hosts periodically
+func (m *AgentSRV) run() {
+
+	for {
+		// Set up a channel and select so we can timeout if findSRV takes too long
+		// set channel to 1 so that goroutines can't pile-up
+		c := make(chan []string, 1)
```

Comment:
Yeah I agree, I had considered that this might be happening (I'm super new at go), but thanks for confirming it. I'll drop it back to a simple loop, and add the startup option you suggest.